### PR TITLE
feat: add Rapid Evaluation SDK tutorial for RAG QA evaluation

### PR DESCRIPTION
**Pull request by jsondai**
# Description
Add RAG QA evaluation notebook tutorial for Vertex AI Python SDK for Rapid Evaluation

Thank you for opening a Pull Request!
Before submitting your PR, there are a few things you can do to make sure it goes smoothly:

- [X] Follow the [`CONTRIBUTING` Guide](https://github.com/GoogleCloudPlatform/generative-ai/blob/main/CONTRIBUTING.md).
- [X] You are listed as the author in your notebook or README file.
  - [X] Your account is listed in [`CODEOWNERS`](https://github.com/GoogleCloudPlatform/generative-ai/blob/main/.github/CODEOWNERS) for the file(s).
- [X] Make your Pull Request title in the <https://www.conventionalcommits.org/> specification.
- [X] Ensure the tests and linter pass (Run `nox -s format` from the repository root to format).
- [X] Appropriate docs were updated (if necessary)

### CODE REVIEW

File: gemini/evaluation/evaluate_generated_answers_from_RAG_for_QA_rapid_evaluation_sdk.ipynb [hunk 1]
```diff
@@ -0,0 +1,589 @@
+{
```
Comment (holtskinner) [inline gemini/evaluation/evaluate_generated_answers_from_RAG_for_QA_rapid_evaluation_sdk.ipynb:1]: <p>I'd recommend renaming this to be more specific about using Rapid Evaluation</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/generative-ai/pull/575/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='1'/>

Comment (holtskinner) [inline gemini/evaluation/evaluate_generated_answers_from_RAG_for_QA_rapid_evaluation_sdk.ipynb:1]: <p>Can you add links to the documentation for this feature?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/generative-ai/pull/575/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='4'/>

Comment (holtskinner) [inline gemini/evaluation/evaluate_generated_answers_from_RAG_for_QA_rapid_evaluation_sdk.ipynb:1]: <p>Add information about restarting kernel. See other notebooks for examples</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/generative-ai/pull/575/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='7'/>

Comment (holtskinner) [inline gemini/evaluation/evaluate_generated_answers_from_RAG_for_QA_rapid_evaluation_sdk.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #2.    %pip install --quiet --upgrade nest_asyncio
</pre><p>Why do you need nest_asyncio ?</p><p> </p><p>I don't see any async code in this notebook. Is it required for the Rapid Evaluation?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/generative-ai/pull/575/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='7'/>

Comment (holtskinner) [inline gemini/evaluation/evaluate_generated_answers_from_RAG_for_QA_rapid_evaluation_sdk.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #18.    import vertexai
</pre><p>Extra import. Also quite a few of these imports aren't used.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/generative-ai/pull/575/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='13'/>

Comment (holtskinner) [inline gemini/evaluation/evaluate_generated_answers_from_RAG_for_QA_rapid_evaluation_sdk.ipynb:1]: <p>Only <code>display_eval_report()</code> and  <code>display_explanations()</code> seem to be used. Can you remove the other functions? </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/generative-ai/pull/575/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='17'/>

Comment (holtskinner) [inline gemini/evaluation/evaluate_generated_answers_from_RAG_for_QA_rapid_evaluation_sdk.ipynb:1]: <p>Are there additional resources describing what each of these metrics do?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/generative-ai/pull/575/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='22'/>

Comment (holtskinner) [inline gemini/evaluation/evaluate_generated_answers_from_RAG_for_QA_rapid_evaluation_sdk.ipynb:1]: <p>Use code font for <code>display_eval_report()</code></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/generative-ai/pull/575/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='27'/>

Comment (holtskinner) [inline gemini/evaluation/evaluate_generated_answers_from_RAG_for_QA_rapid_evaluation_sdk.ipynb:1]: <p>Use code font for <code>display_explanations()</code></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/generative-ai/pull/575/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='29'/>

**Review by holtskinner — CHANGES_REQUESTED** [review]
See comments.

Please add your account to the https://github.com/GoogleCloudPlatform/generative-ai/blob/main/.github/CODEOWNERS for this file.

File: gemini/evaluation/evaluate_generated_answers_from_RAG_for_QA_rapid_evaluation_sdk.ipynb (hunk 1 above)
Comment (jsondai) [reply gemini/evaluation/evaluate_generated_answers_from_RAG_for_QA_rapid_evaluation_sdk.ipynb:1]: <p>Added a link to the public doc.</p>

Comment (jsondai) [reply gemini/evaluation/evaluate_generated_answers_from_RAG_for_QA_rapid_evaluation_sdk.ipynb:1]: <p>I tested this out and the Rapid Eval SDK notebooks do not require this step for restarting the kernel. We've received several comments in previous workshop complaining about it crashing the colab session and causing long delays. Is it necessary to add this?</p>

**Comment by jsondai** [comment]
> See comments.
> 
> Please add your account to the https://github.com/GoogleCloudPlatform/generative-ai/blob/main/.github/CODEOWNERS for this file.

Thanks for reviewing! I am added to the CODEOWNERS for the whole `/generative-ai/gemini/evaluation` path.

File: gemini/evaluation/evaluate_generated_answers_from_RAG_for_QA_rapid_evaluation_sdk.ipynb (hunk 1 above)
Comment (holtskinner) [reply gemini/evaluation/evaluate_generated_answers_from_RAG_for_QA_rapid_evaluation_sdk.ipynb:1]: Ok, shouldn't the SDK itself include that as an import/step for initialization so the user doesn't have to do it?

**Comment by holtskinner** [comment]
@jsondai Understood on the CODEOWNERS.

It looks like your updates based on the review comments weren't added. Can you try again? It's also helpful to keep them as separate commits so we can see when changes based on reviews are made. The commits will all be squashed into one commit when the PR is merged.

**Review by holtskinner — APPROVED** [review]

**Pull request merged**